module/criteria/helper.js

```diff
@@ -459,4 +468,80 @@ module.exports = class criteriaHelper {
     })
 }
 
+
+  /**
+   * Create copy of criterias
+   * @method
+   * @name createCopyOfCriterias
+   * @param {Array} criteriaIds - Array of criteria Id's        
+   * @returns {Object}  old and new Mapped criteria ids .  
+  */
+
+  static createCopyOfCriterias(criteriaIds= []) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        if (!criteriaIds.length) {
+          throw new Error(messageConstants.apiResponses.CRITERIA_ID_REQUIRED)
+        }
+
+        let criteriaDocuments = await this.criteriaDocument
+        (
+          { _id: { $in : criteriaIds }}
+        )
+
+        if (!criteriaDocuments.length) {
+           throw new Error(messageConstants.apiResponses.CRITERIA_NOT_FOUND)
+        }
+
+        let criteriaIdMap = {};
+        
+        await Promise.all(criteriaDocuments.map(async criteria => {
+          
+          await Promise.all(criteria.evidences.map(async evidence => {
```

Comment:
@deepakr123 - Please do for loop here and in sections to avoid proper update of data

Reply:
resolved